member-service/src/main/java/com/hedvig/memberservice/aggregates/MemberAggregate.kt

```diff
@@ -0,0 +1,588 @@
+        if (cmd.phoneNumber != null
+            && member.phoneNumber != cmd.phoneNumber) {
+            apply(PhoneNumberUpdatedEvent(cmd.memberId, cmd.phoneNumber))
+        }
+    }
+
+    @CommandHandler
+    fun on(cmd: SetFraudulentStatusCommand) {
+        apply(FraudulentStatusUpdatedEvent(cmd.memberId, cmd.fraudulentStatus, cmd.fraudulentDescription), MetaData.with("token", cmd.token))
+    }
+
+    @CommandHandler
+    fun on(cmd: UpdateSwedishWebOnBoardingInfoCommand) {
+        logger.debug("Updating ssn and email for webOnBoarding member {}, ssn: {}, email: {}",
+            cmd.memberId, cmd.SSN, cmd.email)
+        if (cmd.SSN != null
+            && member.ssn != cmd.SSN) {
+            apply(SSNUpdatedEvent(id, cmd.SSN, SSNUpdatedEvent.Nationality.SWEDEN))
+            getBirthdateFromSwedishSsnAndApplyEvent(cmd.SSN, null)
+        }
+        if (cmd.email != null
+            && member.email != cmd.email) {
+            apply(EmailUpdatedEvent(id, cmd.email))
+        }
+    }
+
+    @CommandHandler
+    fun on(cmd: UpdateSSNCommand) {
+        logger.debug("Updating ssn for member {}, ssn: {}", cmd.memberId, cmd.ssn)
+        apply(SSNUpdatedEvent(cmd.memberId, cmd.ssn, toSSNUpdatedEventNationality(cmd.nationality)))
+        getBirthdateFromSwedishSsnAndApplyEvent(cmd.ssn, null)
+    }
+
+    @CommandHandler
+    fun on(cmd: InitializeAppleUserCommand) {
+        apply(
+            MemberCreatedEvent(
+                cmd.memberId,
+                MemberStatus.SIGNED)
+        )
+        apply(
+            SSNUpdatedEvent(
+                cmd.memberId,
+                cmd.personalNumber,
+                SSNUpdatedEvent.Nationality.SWEDEN
+            )
+        )
+        getBirthdateFromSwedishSsnAndApplyEvent(cmd.personalNumber, null)
+        apply(
+            NewCashbackSelectedEvent(
+                cmd.memberId,
+                cashbackService.getDefaultId(id).toString())
+        )
+        apply(
+            NameUpdatedEvent(
+                cmd.memberId,
+                cmd.firstName,
+                cmd.lastName),
+            null
+        )
+        apply(
+            PhoneNumberUpdatedEvent(
+                cmd.memberId,
+                cmd.phoneNumber
+            )
+        )
+        apply(
+            EmailUpdatedEvent(
+                cmd.memberId,
+                cmd.email)
+        )
+        apply(LivingAddressUpdatedEvent(
+            cmd.memberId,
+            cmd.street,
+            cmd.city,
+            cmd.zipCode,
+            null,
+            0
+        )
+        )
+    }
+
+    @CommandHandler
+    fun on(cmd: UpdateAcceptLanguageCommand) {
+        logger.info("Updating accept language for member {}, new number: {}", cmd.memberId, cmd.acceptLanguage)
+        if (!cmd.acceptLanguage.isEmpty() &&
+            member.acceptLanguage != cmd.acceptLanguage) {
+            apply(AcceptLanguageUpdatedEvent(cmd.memberId, cmd.acceptLanguage))
+        }
+    }
```

Comment:
I personally prefer logging only when it actually happened. If you want to update it feel free :)
```suggestion
    @CommandHandler
    fun on(cmd: UpdateAcceptLanguageCommand) {
        if (!cmd.acceptLanguage.isEmpty() &&
            member.acceptLanguage != cmd.acceptLanguage) {
            logger.info("Updating accept language for member {}, new number: {}", cmd.memberId, cmd.acceptLanguage)
            apply(AcceptLanguageUpdatedEvent(cmd.memberId, cmd.acceptLanguage))
        }
    }
```